core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -48,15 +48,9 @@ import scala.collection.JavaConverters._
 import org.apache.kafka.common.requests.SaslHandshakeResponse
 
 object KafkaApis {
-  val apiVersionsResponse = new ApiVersionsResponse(Errors.NONE.code, buildApiKeysToApiVersions.values.toList.asJava)
-
-  private def buildApiKeysToApiVersions: Map[Short, ApiVersionsResponse.ApiVersion] = {
-    ApiKeys.values.map(apiKey =>
-      apiKey.id -> new ApiVersionsResponse.ApiVersion(apiKey.id, Protocol.MIN_VERSIONS(apiKey.id), Protocol.CURR_VERSION(apiKey.id))).toMap
-  }
+  val apiVersionsResponse = ApiVersionsResponse.createApiVersionsResponse
```

Comment:
Maybe we should cache this in `ApiVersionsResponse` as a static final field. It's not ideal, but seems like a reasonable compromise given the fact that the `SaslServerAuthenticator` needs access to it.